File: app/src/main/java/com/kent/university/privelt/ui/dashboard/card/data_metrics/DataMetricsViewHolder.java
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.kent.university.privelt.ui.dashboard.card.data_metrics;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

import com.kent.university.privelt.PriVELTApplication;
import com.kent.university.privelt.R;
import com.kent.university.privelt.model.CardItem;

import net.neferett.webviewsextractor.model.UserDataTypes;

public class DataMetricsViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.value_metrics)
    TextView metrics;

    @BindView(R.id.image_type)
    ImageView type;

    DataMetricsViewHolder(@NonNull View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public void bind(CardItem cardItem, boolean isService) {
        metrics.setText(String.valueOf(cardItem.getNumber()));
        if (!isService) {
            UserDataTypes userDataType = UserDataTypes.valueOf(cardItem.getName().toUpperCase());
            type.setImageResource(userDataType.getRes());
        }
        else {
            PriVELTApplication priVELTApplication = (PriVELTApplication) type.getContext().getApplicationContext();
            type.setImageResource(priVELTApplication.getServiceHelper().getResIdWithName(cardItem.getName()));
        }
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/dashboard/card/data_metrics/DataMetricsViewHolder.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.dashboard.card.data_metrics

import android.view.View

import androidx.recyclerview.widget.RecyclerView
import com.kent.university.privelt.PriVELTApplication
import com.kent.university.privelt.events.DetailedCardEvent
import com.kent.university.privelt.model.Card
import com.kent.university.privelt.model.CardItem
import kotlinx.android.synthetic.main.cell_metrics.view.*
import net.neferett.webviewsextractor.model.UserDataTypes
import org.greenrobot.eventbus.EventBus
import java.util.*

class DataMetricsViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
    fun bind(card: Card, cardItem: CardItem) {
        itemView.value_metrics!!.text = cardItem.number.toString()
        if ((card.isService)) {
            val userDataType = UserDataTypes.getUserDataType(cardItem.name.toUpperCase(Locale.ROOT))
            itemView.image_type!!.setImageResource(userDataType.res)
        } else {
            val priVELTApplication = itemView.image_type!!.context.applicationContext as PriVELTApplication
            itemView.image_type!!.setImageResource(priVELTApplication.serviceHelper!!.getResIdWithName(cardItem.name))
        }
        itemView.setOnClickListener { EventBus.getDefault().post(DetailedCardEvent(card)) }
    }
}